README.md

```diff
@@ -0,0 +1,68 @@
+# sysmodule-flake
+sysmodule-flake uses [flake-parts](https://github.com/hercules-ci/flake-parts) to further extend the comfort of using one top-level flake.nix file for all system configurations.
+
+It collects all configurations and modules in the project, found in a pre-defined directory structure (see [Usage](#Usage)). These can be any combination of NixOS configurations, [nix-darwin](https://github.com/LnL7/nix-darwin) configurations and NixOS modules to be imported in the configurations or as [home-manager](https://github.com/nix-community/home-manager) modules. 
+The configuration that applies to the current system is selected and applied automatically during evaluation.
+
+# Usage
+
+Add sysmodule-flake to your configuration's flake inputs and define the `sysmodules-flake` attribute in the outputs as shown in the following example:
+
+```nix
+# flake.nix
+
+{
+  description = "Dummy NixOS (+ nix-darwin + home-manager) config";
+
+  inputs = 
+  {
+    nixpkgs.url = "github:nixos/nixpkgs?ref=nixos-unstable";
+
+    flake-parts.url = "github:hercules-ci/flake-parts";
+    flake-parts.inputs.nixpkgs-lib.follows = "nixpkgs";
+
+    home-manager.url = "github:nix-community/home-manager";
+    home-manager.inputs.nixpkgs.follows = "nixpkgs";
+
+    darwin.url = "github:lnl7/nix-darwin";
```

Comment:
If you called this input `nix-darwin`, it would add the neat side effect that you can later just say `inherit nix-darwin;` instead of `nix-darwin = darwin;`